### ISSUE TITLE
CLI: Inspect on deployment, deamonset and other resources fail 

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#12831]
**Rancher versions:**
rancher/server: Master
Rancher CLI version: rancher-v1.0.0-alpha13

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)** DO 

**Steps to Reproduce:**
1. Create a 2.0 cluster and deploy a workload
2. Setup CLI with the configuration w.r.t the cluster
3. Issue the inspect command for a deployment. It fails with the error below:
```
rancher-v1.0.0-alpha13 soumya$ ./rancher inspect test1 --type deployment
FATA[0000] Bad response statusCode [404]. Status [404 Not Found]. Body: [baseType=error, code=NotFound, message=failed to find resource by id] from [https://167.99.164.121/v3/project/c-n2s5k:project-j2jtr/deployments/deployment:default:test1] 
```
Inspect on Deamonset, stateful set and cronjob fail as below:
```
LAPTOP-6CQEFUBT:rancher-v1.0.0-alpha13 soumya$ ./rancher inspect globaltest --type daemonsets
FATA[0002] unkown resource type                         
LAPTOP-6CQEFUBT:rancher-v1.0.0-alpha13 soumya$ ./rancher inspect statefultest --type Statefulset
FATA[0001] unkown resource type                         
LAPTOP-6CQEFUBT:rancher-v1.0.0-alpha13 soumya$ ./rancher inspect testcron --type cronjob
FATA[0001] unkown resource type                         

```

**Comment by dramich** [comment]
This is an issue with the server, see https://github.com/rancher/rancher/issues/12837

**Comment by dramich** [comment]
Server issue has been addressed and this works as expected in the cli

**Comment by soumyalj** [comment]
Tested with master. inspect works on these resources when no type is specified. If a type is specified it is case sensitive/has to be in some specific case which the api supports
```
LAPTOP-6CQEFUBT:rancher-v1.0.0-alpha13 soumya$  ./rancher inspect testcron --type cronjob
FATA[0000] unkown resource type       

LAPTOP-6CQEFUBT:rancher-v1.0.0-alpha13 soumya$ ./rancher inspect daemonset1 --type daemonset
FATA[0000] unkown resource type 

LAPTOP-6CQEFUBT:rancher-v1.0.0-alpha13 soumya$ ./rancher inspect daemonset1 --type statefulset
FATA[0000] unkown resource type      

```

Note: kubectl is NOT case sensitive for any of these resources.

**Comment by deniseschannon** [comment]
Available with CLI v2.0.1-rc2

**Comment by bmdepesa** [comment]
rancher/rancher:master 5/15
CLI v2.0.1-rc2 

In cli type flag on inspect is case insensitive.
